Product.cs

```diff
@@ -20,41 +22,40 @@ public double Price
             }
         }
 
-        private static double _taxPercentage = 0.2;
-        public static double TaxPercentage
-        {
-            get => _taxPercentage;
-            set
-            {
-                if (value < 0)
-                {
-                    throw new ArgumentException("Tax percentage cannot be a negative value.");
-                }
-
-                _taxPercentage = value;
-            }
-        }
+        public static PriceCalculator priceCalculator = new PriceCalculator();

```

Comment:
Imagine I have 4 products, and I wanted to change the tax percentage, or the discount percentage.
I need to iterate over all products and change it for them. Is it practical?

Reply:
in the requirements the tax an discount is equal for all products

Reply:
@Mo7ammadAbuSafat Ok so I see you've added the price calculator as a static field.